### PR TITLE
logictest: use v22_2 schema in mixed version test

### PR DESCRIPTION
**Pull request by jeffswenson**
Previously, the v22.2-v23.1 mixed version test was using the v22.2 version gate with a v23.1 schema. This caused the alter_table test to fail on PR #98446. The lease protocol only works if the version gates and the schema are in sync.

As part of this change I simplified the version configuration in the logic tests so that the only version key is the bootstrap version. It doesn't really make sense to specify a binary version different than the build version. The binary will always have the build version's behavior.

This change also deletes the local-v1.1-at-v1.0-noupgrade test. The version can't be described by a version key and the tests do not represent a situation that occurs in real world deployments.

Part of #94843

Release note: None

### CODE REVIEW

File: pkg/sql/logictest/testdata/logic_test/alter_table [hunk 1]
```diff
@@ -1,5 +1,4 @@
 # LogicTest: default-configs local-mixed-22.2-23.1
-
```
Comment (ajwerner) [inline pkg/sql/logictest/testdata/logic_test/alter_table:2]: Nit keep this

**Review by ajwerner — APPROVED** [review]
Nice

**Review by michae2 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98695#-:-NQajjUwDABVhuUTVvgM:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @JeffSwenson)

___
*[`pkg/sql/logictest/logictestbase/logictestbase.go` line 59 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98695#-NQaj5R48ht43p9NVAuI:-NQaj5R5AjkQT8h-rIiL:b-8v9y63) ([raw file](https://github.com/cockroachdb/cockroach/blob/16253cef4be39e0cf8085149194858b4a55849f2/pkg/sql/logictest/logictestbase/logictestbase.go#L59)):*
> ```Go
> 	// bootstrapped at.
> 	BootstrapVersion clusterversion.Key
> 	// DisableUpgrade prevents the cluster from automatically ruinng upgrading
> ```

typo: delete "ruinng"

**Comment by JeffSwenson** [comment]
bors -

**Pull request merged**